packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -178,6 +178,18 @@ class RequestWrapper {
   async fetch({request}) {
     assert.atLeastOne({request});
 
+    if (this.behaviorCallbacks.requestWillFetch) {
+      for (let callback of this.behaviorCallbacks.requestWillFetch) {
+        const returnedRequest = callback({request});
+        if (returnedRequest && returnedRequest.then) {
```

Comment:
doesn't webP plugin already had one? saving device's pixel ratio/DPR in IDB and using it while modifying it in IDB?